doc/source/data/key-concepts.rst

```diff
@@ -10,19 +10,20 @@ Key Concepts
 Datasets
 --------
 
-A Dataset consists of a list of Ray object references to *blocks*.
-Each block holds a set of items in either `Arrow table format <https://arrow.apache.org/docs/python/data.html#tables>`__
-or a Python list (for non-tabular data).
-For ML use cases, Datasets also natively supports mixing :ref:`Tensor <datasets_tensor_support>` and tabular data.
+A :term:`Dataset` contains a list of Ray object references to :term:`blocks <Block>`.
+Each block holds a set of items in a `Arrow table <https://arrow.apache.org/docs/python/data.html#tables>`_,
+`pandas DataFrame <https://pandas.pydata.org/docs/reference/frame.html>`_, or Python list.
```

Comment:
capitalize for consistency with Arrow table? 
```suggestion
`Pandas DataFrame <https://pandas.pydata.org/docs/reference/frame.html>`_, or Python list.
```

Reply:
I don't think. pandas doesn't capitalized the name in their own documentation, whereas Arrow does.